packages/flutter_tools/test/general.shard/resident_runner_test.dart

```diff
@@ -2703,6 +2703,38 @@ void main() {
     }) async => mockVMService,
   }));
 
+  testUsingContext('Failed DDS start outputs error message', () => testbed.run(() async {
+    // See https://github.com/flutter/flutter/pull/72736 for context.
+    fakeVmServiceHost = FakeVmServiceHost(requests: <VmServiceExpectation>[]);
+    final MockDevice mockDevice = MockDevice();
+    when(mockDevice.dds).thenReturn(DartDevelopmentService(logger: testLogger));
+    ddsLauncherCallback = (Uri uri, {bool enableAuthCodes, bool ipv6, Uri serviceUri}) {
+      throw FakeDartDevelopmentServiceException(message: 'No URI');
+    };
+
+    final TestFlutterDevice flutterDevice = TestFlutterDevice(
+      mockDevice,
+      observatoryUris: Stream<Uri>.value(testUri),
+    );
+    bool caught = false;
+    final Completer<void>done = Completer<void>();
+    runZonedGuarded(() {
```

Comment:
If the above is changed to `handleError(e)` from `rethrow`, can this be:
```dart
try {
  await flutterDevice.connect(...);
} on dds.DartDevelopmentServiceException catch (e) {
  ...
}
```

Reply:
Ah, that's much nicer!